src/Nox.Integration/Services/NoxIntegration.cs

```diff
@@ -46,33 +50,60 @@ public NoxIntegration(ILogger logger, Solution.Integration definition)
         }
     }
     
-    public async Task<bool> ExecuteAsync()
+    public async Task<bool> ExecuteAsync(IEnumerable<INoxCustomTransformHandler>? handlers = null)
     {
         var result = false;
 
         switch (MergeType)
         {
             case IntegrationMergeType.MergeNew:
-                result = await ExecuteMergeNew();
+                result = await ExecuteMergeNew(handlers);
                 break;
             case IntegrationMergeType.AddNew:
-                result = await ExecuteAddNew();
+                result = await ExecuteAddNew(handlers);
                 break;
         }
 
         return result;
     }
 
-    private async Task<bool> ExecuteMergeNew()
+    private async Task<bool> ExecuteMergeNew(IEnumerable<INoxCustomTransformHandler>? handlers)
     {
         var source = ReceiveAdapter!.DataFlowSource;
         //todo: filter source using merge state
 
+
+        IDataFlowSource<ExpandoObject>? transformSource = null;
+        
+        if (TransformType == IntegrationTransformType.CustomTransform)
+        {
+            if (handlers == null) throw new NoxIntegrationException("Cannot execute custom transform, no handlers registered.");
+            //Find the handler
+            try
+            {
+                var handler = handlers.Single(h => h.IntegrationName == Name);
```

Comment:
IMO the handler can be selected on NoxContext like the  NoxIntegration is
NoxIntegration could only know ist proper handler